libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineViewTest.kt

```diff
@@ -311,109 +360,508 @@ class MapboxRouteLineViewTest {
         val primaryRouteCasingLayer = mockk<LineLayer>(relaxed = true)
         val restrictedLineLayer = mockk<LineLayer>(relaxed = true)
         val topLevelLayer = mockk<LineLayer>(relaxed = true)
+        val primaryTrailLayer = mockk<LineLayer>(relaxed = true)
+        val primaryTrailCasingLayer = mockk<LineLayer>(relaxed = true)
         val style = mockk<Style> {
             every {
-                getLayer(PRIMARY_ROUTE_TRAFFIC_LAYER_ID)
+                getLayer(LAYER_GROUP_1_TRAFFIC)
             } returns primaryRouteTrafficLayer
             every {
-                getLayer(PRIMARY_ROUTE_LAYER_ID)
+                getLayer(LAYER_GROUP_1_MAIN)
             } returns primaryRouteLayer
             every {
-                getLayer(PRIMARY_ROUTE_CASING_LAYER_ID)
+                getLayer(LAYER_GROUP_1_CASING)
             } returns primaryRouteCasingLayer
             every {
-                getLayer(RESTRICTED_ROAD_LAYER_ID)
+                getLayer(LAYER_GROUP_1_RESTRICTED)
             } returns restrictedLineLayer
             every {
                 getLayer(TOP_LEVEL_ROUTE_LINE_LAYER_ID)
             } returns topLevelLayer
+            every {
+                getLayer(LAYER_GROUP_1_TRAIL)
+            } returns primaryTrailLayer
+            every {
+                getLayer(LAYER_GROUP_1_TRAIL_CASING)
+            } returns primaryTrailCasingLayer
         }.also {
             mockCheckForLayerInitialization(it)
         }
 
         pauseDispatcher {
-            MapboxRouteLineView(options).renderRouteLineUpdate(style, state)
+            val view = MapboxRouteLineView(options)
+            view.initPrimaryRouteLineLayerGroup(MapboxRouteLineUtils.layerGroup1SourceLayerIds)
+            view.renderRouteLineUpdate(style, state)
             verify { MapboxRouteLineUtils.initializeLayers(style, options) }
         }
 
         verify { primaryRouteTrafficLayer.lineTrimOffset(trafficLineExp) }
         verify { primaryRouteLayer.lineTrimOffset(routeLineExp) }
         verify { primaryRouteCasingLayer.lineTrimOffset(casingLineEx) }
         verify { restrictedLineLayer.lineTrimOffset(restrictedRoadExp) }
+        verify { primaryTrailLayer.lineTrimOffset(trailExp) }
+        verify { primaryTrailCasingLayer.lineTrimOffset(trailCasingExp) }
+        unmockkObject(MapboxRouteLineUtils)
+        unmockkStatic("com.mapbox.maps.extension.style.layers.LayerUtils")
+    }
+
+    @Test
+    fun `renderRouteDrawData when new routes`() = coroutineRule.runBlockingTest {
```

Comment:
The has to be an opportunity to simplify below tests, there's a lot of redundancy.

Reply:
I agree but it wasn't straight forward. I'd like to revisit this at a later time.